File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

public class Solution {
    public Integer solution(int[] priorities, int location) {
        PriorityQueue<Integer> q = new PriorityQueue<>(Collections.reverseOrder()); //우선순위 큐를 내림차순 정렬로 설정한다.
        int answer = 0;

        for (int priority : priorities) {
            q.add(priority);
        }

        while (!q.isEmpty()) {
            for (int i = 0; i < priorities.length; i++) {
                if (priorities[i] == q.peek()) {
                    if (i == location) {
                        answer++;
                        return answer;
                    }
                    q.poll();
                    answer++;
                }
            }
        }
        return -1;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    
    priorities = [[i,p] for i,p in enumerate(priorities)]
    
    while priorities:
        now = priorities.pop(0)
        if any(p[1] > now[1] for p in priorities):
            priorities.append(now)
        else:
            if now[0] == location: 
                return answer+1
            answer+=1
    return -1

